### PR TITLE
Updated qtox to the last version

### PR DESCRIPTION
**Pull request by akaWolf**
For adding the latest qtox version, I've splitted qtox and filteraudio to different packages and updated libtoxcore.
Since .pro file of qtox has hardcoded paths, I've added NIX_CFLAGS_COMPILE with appropriate content.

### CODE REVIEW

**Comment by geerds** [comment]
You need to test every single package that uses libtoxtore

**Comment by akaWolf** [comment]
Looks like ratox isn't possible to build with the latest libtoxcore.

**Comment by akaWolf** [comment]
Will we need two versions of libtoxcore?

**Comment by lethalman** [comment]
That's the kind of project that makes me cry. Start by complaining upstream forever about their hardcoded paths and how they break tens of years of portability.

**Comment by lethalman** [comment]
@akaWolf also please, please, fill `NIX_CFLAGS_COMPILE` with `pkg-config --cflags` and not hardcoded like you did, as well.

**Comment by akaWolf** [comment]
@lethalman I saw a tens of examples of using NIX_CFLAGS_COMPILE in nixpkgs in the same way, as I did. Why I need to use pkg-config here?

**Comment by lethalman** [comment]
@akaWolf *broken* examples of using NIX_CFLAGS_COMPILE, or perhaps for libraries that didn't have a proper pc file.

If you don't want to use pkg-config feel free to keep the PR as is, @geerds will probably merge it anyway.

**Comment by akaWolf** [comment]
@lethalman it's better to solve a problem with ratox and toxprpl. And after I can do something for PR.
I'm thinking about having a two different versions of libtoxcore.

**Comment by jagajaga** [comment]
And don't forget to rename commits as it said in `CONTRIBUTING.md`.

**Comment by akaWolf** [comment]
I've added new-api version of libtoxcore and leave old-api version, since at least 2 packages doesn't support new api.

**Comment by akaWolf** [comment]
And changed include paths to using pkg-config, according to @lethalman notice.

File: pkgs/applications/networking/instant-messengers/qtox/default.nix [hunk 1]
```diff
@@ -1,41 +1,38 @@
...
-      qt5.base qt5.tools libXScrnSaver
+      libtoxcore-dev openal opencv libsodium filter-audio
+      qt5.base qt5.tools libXScrnSaver glib gtk2 cairo
+      pango atk qrencode ffmpeg qt5.translations
     ];
+
   nativeBuildInputs = [ pkgconfig ];
 
-  configurePhase = "qmake";
+    preConfigure = ''
```
Comment (jagajaga) [inline pkgs/applications/networking/instant-messengers/qtox/default.nix]: Seems that there is an extra space..

File: pkgs/development/libraries/filter-audio/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{ stdenv, fetchgit }:
+
+let
+  version = "612c5a102550c614e4c8f859e753ea64c0b7250c";
+  date = "20150624";
+in
+stdenv.mkDerivation rec {
+  name = "filter-audio-${date}-${builtins.substring 0 7 version}";
```
Comment (jagajaga) [inline pkgs/development/libraries/filter-audio/default.nix]: Why do you add revision?

File: pkgs/development/libraries/libtoxcore/new-api/default.nix [hunk 3]
```diff
@@ -0,0 +1,63 @@
+{ stdenv, fetchFromGitHub, autoconf, libtool, automake, libsodium, ncurses, libopus
+, libvpx, check, libconfig, pkgconfig }:
+
+let
+  version = "219fabc0f5dbaac7968cb7728d25dface3ebb2ea";
+  date = "20150629";
+in
+stdenv.mkDerivation rec {
+  name = "tox-core-dev-${date}-${builtins.substring 0 7 version}";
```
Comment (jagajaga) [inline pkgs/development/libraries/libtoxcore/new-api/default.nix]: Same here.

File: pkgs/applications/networking/instant-messengers/qtox/default.nix [hunk 4]
```diff
@@ -1,41 +1,38 @@
-{ stdenv, fetchFromGitHub, pkgconfig, libtoxcore, qt5, openal, opencv
-, libsodium, libXScrnSaver }:
+{ stdenv, fetchgit, pkgconfig, libtoxcore-dev, qt5, openal, opencv,
+  libsodium, libXScrnSaver, glib, gdk_pixbuf, gtk2, cairo,
+  pango, atk, qrencode, ffmpeg, filter-audio }:
 
-let
+stdenv.mkDerivation rec {
+  name = "qtox-dev-20150624";
```
Comment (jagajaga) [inline pkgs/applications/networking/instant-messengers/qtox/default.nix]: Better name it as `qtox-git-date`.

Comment (akaWolf) [inline pkgs/applications/networking/instant-messengers/qtox/default.nix]: I don't understand, why we even need qtox-something, not just qtox.

File: pkgs/development/libraries/filter-audio/default.nix (hunk 2 above)
Comment (akaWolf) [inline pkgs/development/libraries/filter-audio/default.nix]: I saw that in one of a package. I can delete revision.

File: pkgs/applications/networking/instant-messengers/qtox/default.nix (hunk 4 above)
Comment (jagajaga) [inline pkgs/applications/networking/instant-messengers/qtox/default.nix]: Wait, it was previously named as `qtox-dev-date`? If so leave as it was.

File: pkgs/development/libraries/filter-audio/default.nix (hunk 2 above)
Comment (jagajaga) [inline pkgs/development/libraries/filter-audio/default.nix]: Yes, revisions are not needed due to `date`.

File: pkgs/applications/networking/instant-messengers/qtox/default.nix (hunk 4 above)
Comment (akaWolf) [inline pkgs/applications/networking/instant-messengers/qtox/default.nix]: Yes, it's so.

File: pkgs/development/libraries/filter-audio/default.nix (hunk 2 above)
Comment (akaWolf) [inline pkgs/development/libraries/filter-audio/default.nix]: Ok.

File: pkgs/applications/networking/instant-messengers/qtox/default.nix (hunk 1 above)
Comment (akaWolf) [inline pkgs/applications/networking/instant-messengers/qtox/default.nix]: Fixed.

**Pull request merged**